### PR TITLE
Add axis constraints on grow; align produces reference; refactor to, at, cutout.

### PR DESCRIPTION
**Pull request by pentacular**

### CODE REVIEW

File: nb/documentation/creating_geometry/creating_geometry.md [hunk 1]
```diff
@@ -70,7 +70,7 @@ Hexagon(6, 3, 2).view();
 A built in single line font useful for adding text
 
 ```JavaScript
-Hershey('Some Example Text', 20).align('xy').view();
+Hershey('Some Example Text', 20).to(align('xy')).view();
```
Comment (BarbourSmith) [inline nb/documentation/creating_geometry/creating_geometry.md:73]: 👍

File: algorithm/cgal/cgal.cc [hunk 2]
```diff
@@ -4934,13 +5013,14 @@ int Link(Geometry* geometry, bool close) {
   return STATUS_OK;
 }
 
-int Loft(Geometry* geometry) {
+int Loft(Geometry* geometry, bool close) {
```
[the remark this answers is not in the document]
Comment (pentacular) [reply algorithm/cgal/cgal.cc:5016]: It runs a hole-filling pass at the end -- generally this should be set true at the api level.

[the remark this answers is not in the document]
Comment (pentacular) [reply algorithm/cgal/cgal.cc:5016]: Rather, normal operations should probably be setting this true at the api level -- producing unclosed lofts should be unusual.

File: algorithm/cgal/grow.js [hunk 3]
```diff
@@ -4,11 +4,14 @@ import { fromCgalGeometry, withCgalGeometry } from './cgalGeometry.js';
 import { ErrorZeroThickness } from './error.js';
 import { toCgalTransformFromJsTransform } from './transform.js';
 
-export const grow = (inputs, offset) =>
+export const grow = (inputs, offset, { x = true, y = true, z = true } = {}) =>
```
[the remark this answers is not in the document]
Comment (pentacular) [reply algorithm/cgal/grow.js:7]: Yeah, I've wanted this for a while.

File: nb/projects/saucer/saucer.nb [hunk 4]
```diff
@@ -1,21 +1,82 @@
...
+    s
+      .get('top')
+      .rx(1 / 2)
+      .to(align('z>'))
+      .on(get('peg'), (s) => Empty())
+  )
+  .stl('base', (s) =>
+    s
+      .get('base')
+      .to(align('z>'))
```
[the remark this answers is not in the document]
Comment (pentacular) [reply nb/projects/saucer/saucer.nb:71]: I think the move to references generally makes sense.

I'm a lot happier with the cleaned up .to and .at semantics.

.to moves something relative to a reference point
.at performs an operation on something relative to a reference point, but doesn't move it.

**Pull request merged**